management/branding.go

```diff
@@ -16,10 +20,102 @@ type Branding struct {
 type BrandingColors struct {
 	// Accent color
 	Primary *string `json:"primary,omitempty"`
+
 	// Page background color
+	// Only one of PageBackground and PageBackgroundGradient should be set.
+	// If both fields are set, PageBackground takes priority.
+	PageBackground *string `json:"-"`
+
+	// Page background gradient
+	// Only one of PageBackground and PageBackgroundGradient should be set.
+	// If both fields are set, PageBackground takes priority.
+	PageBackgroundGradient *BrandingPageBackgroundGradient `json:"-"`
+}
+
+type BrandingPageBackgroundGradient struct {
+	Type        *string `json:"type,omitempty"`
+	Start       *string `json:"start,omitempty"`
+	End         *string `json:"end,omitempty"`
+	AngleDegree *int    `json:"angle_deg,omitempty"`
+}
+
+type jsonPageBackgroundSolid struct {
 	PageBackground *string `json:"page_background,omitempty"`
 }
 
+type jsonPageBackgroundGradient struct {
+	PageBackgroundGradient *BrandingPageBackgroundGradient `json:"page_background,omitempty"`
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+//
+// It is required to handle the json field page_background, which can either
+// be a hex color string, or an object describing a gradient.
+func (bc *BrandingColors) MarshalJSON() ([]byte, error) {
+	var data interface{}
+
+	if bc.PageBackground != nil {
+		data = struct {
+			Primary *string `json:"primary,omitempty"`
+			*jsonPageBackgroundSolid
+		}{
+			Primary: bc.Primary,
+			jsonPageBackgroundSolid: &jsonPageBackgroundSolid{
+				PageBackground: bc.PageBackground,
+			},
+		}
+	} else if bc.PageBackgroundGradient != nil {
+		data = struct {
+			Primary *string `json:"primary,omitempty"`
+			*jsonPageBackgroundGradient
+		}{
+			Primary: bc.Primary,
+			jsonPageBackgroundGradient: &jsonPageBackgroundGradient{
+				PageBackgroundGradient: bc.PageBackgroundGradient,
+			},
+		}
+	}
+
+	return json.Marshal(data)
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+//
+// It is required to handle the json field page_background, which can either
+// be a hex color string, or an object describing a gradient.
+func (bc *BrandingColors) UnmarshalJSON(data []byte) error {
+	// Unmarshal common fields
+	type JSONBrandingColors BrandingColors
```

Comment:
By defining a type alias we can unmarshal the common fields (`Primary`) without running into a stack overflow.